### DISCUSSION TITLE
How to replicate "Parameters in the body"??

### DISCUSSION THREAD

**Discussion by jtomek** [vlang/v#14032] [General]
Hello! I'm new to vlang and love it. I wondered how to replicate the following "parameters in the body" functionality for a custom struct? 

Example: 

```v
sql db {
    create table Foo
}
```

How to make this code executable for `struct Something{}`?

```
// maybe some params here [??]

struct Something{

   // maybe some code here??

}

// maybe some fn that enriches the struct here???
```

So I can call it like 

```
Something{
    create table Foo
}
```

I have spent 24 hours reading the documentation and can't figure it out. My candidates are:
- if I pass a string, it's converted into the first struct parameter
- something with interfaces (still paying around), reviewed orm.v and the @select, create seems promising
- something with the [params], still playing around

Can you please push me to the right direction???

**Comment by StunxFS** [comment]
@jtomek  `sql` is just a expression, like `if` or `match`.

**Comment by jtomek** [comment]
@StunxFS, I see. 
And is it possible to extend the language and create my own expressions? I guess the `sql` expression is written in a v module somewhere?

**Reply by StunxFS** [reply]
Nop
